namer/marathon/src/main/scala/io/buoyant/namer/marathon/MarathonInitializer.scala

```diff
@@ -71,6 +74,28 @@ object MarathonSecret {
       throw Invalid(s)
   }
 
+  def mkAuthenticated(s: MarathonSecret, params: Stack.Params): Authenticator.Authenticated = s match {
+    case MarathonSecret(Some(loginEndpoint), Some(privateKey), Some("RS256"), Some(uid)) =>
+      val loginUrl = new URI(loginEndpoint)
+
+      val port = loginUrl.getPort
+
+      val dest = if (port == -1) s"${loginUrl.getHost}:80" else s"${loginUrl.getHost}:$port"
```

Comment:
I don't think we need to manually add the default port, I think finagle will take care of this.  Also, wouldn't the default port be 443 for https?

Reply:
Surprisingly, it does not take care of it, but we can use the .defaultPort() method on URL to get the proper port in case its not specified. The Marathon docs do not impose specifying a port on the login_endpoint URL: https://jenkins.io/doc/pipeline/steps/marathon/


```
java.lang.IllegalArgumentException: requirement failed: You must specify host and port in hosts: google.com
	at scala.Predef$.require(Predef.scala:277)
	at com.twitter.finagle.util.InetSocketAddressUtil$.$anonfun$parseHostPorts$3(InetSocketAddressUtil.scala:38)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:193)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:193)
	at com.twitter.finagle.util.InetSocketAddressUtil$.parseHostPorts(InetSocketAddressUtil.scala:37)
	at com.twitter.finagle.InetResolver.$anonfun$bind$1(InetResolver.scala:166)
	at com.twitter.util.Try$.apply(Try.scala:26)
	at com.twitter.finagle.InetResolver.bind(InetResolver.scala:166)
	at com.twitter.finagle.BaseResolver.eval(Resolver.scala:171)
	at com.twitter.finagle.BaseResolver.evalLabeled(Resolver.scala:187)
	at com.twitter.finagle.Client.newService(Client.scala:42)
	at com.twitter.finagle.Client.newService$(Client.scala:41)
	at com.twitter.finagle.Http$Client.newService(Http.scala:224)
	at io.buoyant.namer.marathon.MarathonSecret$.mkAuthenticated(MarathonInitializer.scala:90)
	at io.buoyant.namer.marathon.MarathonConfig.newNamer(MarathonInitializer.scala:176)
	at io.buoyant.namer.marathon.MarathonConfig.newNamer(MarathonInitializer.scala:129)
	at io.buoyant.namer.NamerConfig.mk(NamerInitializer.scala:58)
	at io.buoyant.linkerd.Linker$LinkerConfig.$anonfun$mkNamers$2(Linker.scala:180)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:378)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.immutable.List.map(List.scala:284)
	at io.buoyant.linkerd.Linker$LinkerConfig.mkNamers(Linker.scala:174)
	at io.buoyant.linkerd.Linker$LinkerConfig.mk(Linker.scala:151)
	at io.buoyant.linkerd.Main$.main(Main.scala:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.twitter.app.App.$anonfun$nonExitingMain$4(App.scala:364)
	at scala.Option.foreach(Option.scala:257)
	at com.twitter.app.App.nonExitingMain(App.scala:363)
	at com.twitter.app.App.nonExitingMain$(App.scala:344)
	at io.buoyant.linkerd.Main$.nonExitingMain(Main.scala:21)
	at com.twitter.app.App.main(App.scala:333)
	at com.twitter.app.App.main$(App.scala:331)
	at io.buoyant.linkerd.Main$.main(Main.scala:21)
	at io.buoyant.linkerd.Main.main(Main.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sbt.Run.invokeMain(Run.scala:67)
	at sbt.Run.run0(Run.scala:61)
	at sbt.Run.sbt$Run$$execute$1(Run.scala:51)
	at sbt.Run$$anonfun$run$1.apply$mcV$sp(Run.scala:55)
	at sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	at sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	at sbt.Logger$$anon$4.apply(Logger.scala:84)
	at sbt.TrapExit$App.run(TrapExit.scala:248)
	at java.lang.Thread.run(Thread.java:748)
```